src/app/cli/tool-confirmation/cli-confirmation-handler.ts

```diff
@@ -0,0 +1,202 @@
+import { logger } from '../../../core/logger/index.js';
+import * as readline from 'readline';
+import chalk from 'chalk';
+import { InMemorySettingsProvider } from '../../../core/settings/in-memory-provider.js';
+import { SettingsProvider } from '../../../core/settings/types.js';
+import boxen from 'boxen';
+import {
+    ToolConfirmationEvent,
+    ToolConfirmationResponse,
+} from '../../../core/client/tool-confirmation/types.js';
+import { EventBasedConfirmationProvider } from '../../../core/client/tool-confirmation/event-based-confirmation-provider.js';
+
+/**
+ * CLI-specific handler for tool confirmation events
+ * Listens to events from EventBasedConfirmationProvider and handles CLI-specific user interaction
+ */
+export class CLIConfirmationHandler {
+    private settingsProvider: SettingsProvider;
+    private provider: EventBasedConfirmationProvider;
+
+    constructor(provider: EventBasedConfirmationProvider, settingsProvider?: SettingsProvider) {
+        this.provider = provider;
+        this.settingsProvider = settingsProvider ?? new InMemorySettingsProvider();
+
+        // Ensure tool approval is required by default in CLI
+        void this.settingsProvider.updateUserSettings('default', { toolApprovalRequired: true });
+
+        // Listen to confirmation request events
+        this.provider.on('toolConfirmationRequest', this.handleConfirmationRequest.bind(this));
```

Comment:
@coderabbitai this event should be on the agent event bus and handled in the subscriber that the respective UI creates